src/Tools/Source/CompilerGeneratorTools/Source/CSharpSyntaxGenerator/SourceWriter.cs

```diff
@@ -591,140 +572,134 @@ private void WriteGreenRewriter()
                     if (nWritten > 0)
                         WriteLine();
                     nWritten++;
-                    WriteLine("    public override CSharpSyntaxNode Visit{0}({1} node)", StripPost(node.Name, "Syntax"), node.Name);
-                    WriteLine("    {");
-                    for (int f = 0; f < nodeFields.Count; f++)
+                    WriteLine("public override CSharpSyntaxNode Visit{0}({1} node)", StripPost(node.Name, "Syntax"), node.Name);
+                    Indent();
+
+                    if (nodeFields.Count == 0)
                     {
-                        var field = nodeFields[f];
-                        if (IsAnyList(field.Type))
-                        {
-                            WriteLine("      var {0} = this.VisitList(node.{1});", CamelCase(field.Name), field.Name);
-                        }
-                        else
-                        {
-                            WriteLine("      var {0} = ({1})this.Visit(node.{2});", CamelCase(field.Name), field.Type, field.Name);
-                        }
+                        WriteLine("=> node;");
                     }
-                    if (nodeFields.Count > 0)
+                    else
                     {
-                        Write("      return node.Update(");
+                        Write("=> node.Update(");
```

Comment:
simple functional updates/withers/etc. just moved to inline their values instead of needing to explicitly make all the temporaries.